src/widget/wfindonwebmenu.cpp

```diff
@@ -0,0 +1,177 @@
+
+const QString kSearchUrlLastFmTitle = QStringLiteral("https://www.last.fm/search/tracks?");
+
+const QString kSearchUrlLastFmAlbum = QStringLiteral("https://www.last.fm/search/albums?");
+
+const QString kSearchUrlDefault = QStringLiteral("https://soundcloud.com/search?");
+
+QString getServiceUrl(WFindOnWebMenu::Service service,
+        WFindOnWebMenu::TrackSearchProperties trackSearchProperty) {
+    switch (service) {
+    case WFindOnWebMenu::Service::Discogs:
+        return kSearchUrlDiscogsGen;
+    case WFindOnWebMenu::Service::LastFm:
+        if (trackSearchProperty == WFindOnWebMenu::TrackSearchProperties::Title ||
+                trackSearchProperty == WFindOnWebMenu::TrackSearchProperties::ArtistAndTitle) {
+            return kSearchUrlLastFmTitle;
+        } else if (trackSearchProperty == WFindOnWebMenu::TrackSearchProperties::Artist) {
+            return kSearchUrlLastFmArtist;
+        } else {
+            return kSearchUrlLastFmAlbum;
+        }
+    case WFindOnWebMenu::Service::SoundCloud:
+        if (trackSearchProperty == WFindOnWebMenu::TrackSearchProperties::Title ||
+                trackSearchProperty == WFindOnWebMenu::TrackSearchProperties::ArtistAndTitle) {
+            return kSearchUrlSoundCloudTitle;
+        } else if (trackSearchProperty == WFindOnWebMenu::TrackSearchProperties::Artist) {
+            return kSearchUrlSoundCloudArtist;
+        } else {
+            return kSearchUrlSoundCloudAlbum;
+        }
+    default:
+        return kSearchUrlDefault;
+    }
+}
+} // namespace
+
+WFindOnWebMenu::WFindOnWebMenu(QWidget* parent)
+        : QMenu(tr("Find on Web"), parent) {
+}
+
+void WFindOnWebMenu::addSubmenusForServices(const Track& track) {
+    WFindOnWebMenu::populateFromTrackProperties(
+            track,
+            QString(tr("SoundCloud")),
+            Service::SoundCloud);
+    WFindOnWebMenu::populateFromTrackProperties(
+            track, QString(tr("LastFm")), Service::LastFm);
+    WFindOnWebMenu::populateFromTrackProperties(
+            track, QString(tr("Discogs")), Service::Discogs);
+}
+
+void WFindOnWebMenu::addActions(Service service,
+        const QString& trackProperty,
+        QMenu* pServiceMenu,
+        TrackSearchProperties trackSearchProperty) {
+    const QString prefixAction = composePrefixAction(trackSearchProperty);
+    pServiceMenu->addAction(
+            composeActionText(prefixAction, trackProperty),
+            this,
+            [this, service, trackSearchProperty, trackProperty]() {
+                emit triggerBrowser(service, trackSearchProperty, trackProperty);
+            });
+}
+
+void WFindOnWebMenu::openInBrowser(Service service,
+        TrackSearchProperties trackSearchProperty,
+        const QString& query) {
+    const QString serviceUrl = getServiceUrl(service, trackSearchProperty);
+    QUrlQuery urlQuery;
+    urlQuery.addQueryItem("q", query);
```

Comment:
query is not the whole query, it is only the queryValue, or trackPropertyValue or ...